com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationMetadataBuilder.java

```diff
@@ -0,0 +1,80 @@
+
+        List<String> providedDependenciesNames = module.getProvidedDependencies()
+                                                       .stream()
+                                                       .filter(ProvidedDependency::isPublic)
+                                                       .map(ProvidedDependency::getName)
+                                                       .collect(Collectors.toList());
+
+        DeployedMtaModule deployedMtaModule = DeployedMtaModule.builder()
+                                                               .withModuleName(module.getName())
+                                                               .withAppName(NameUtil.getApplicationName(module))
+                                                               .withProvidedDependencyNames(providedDependenciesNames)
+                                                               .withResources(deployedResources)
+                                                               .withUris(uris)
+                                                               .build();
+
+        return Metadata.builder()
+                       .label(MtaMetadataLabels.MTA_ID, deploymentDescriptor.getId())
+                       .label(MtaMetadataLabels.MTA_VERSION, deploymentDescriptor.getVersion())
+                       .annotation(ApplicationMtaMetadataExtractor.MODULE, JsonUtil.toJson(deployedMtaModule, true))
+                       .build();
+    }
+
+    private static DeployedMtaResource mapResourceToDeployedMtaResource(ResourceAndResourceType applicationService, Module module) {
+        ResourceType resourceType = applicationService.getResourceType();
+        Resource resource = applicationService.getResource();
+        if (resourceType != ResourceType.USER_PROVIDED_SERVICE) {
+            return DeployedMtaResource.builder()
+                                      .withServiceName(NameUtil.getServiceName(resource))
+                                      .withResourceName(resource.getName())
+                                      .build();
+        }
+
+        List<DeployedMtaModule> deployedMtaModules = Collections.singletonList(DeployedMtaModule.builder()
+                                                                                                .withModuleName(module.getName())
+                                                                                                .withAppName(NameUtil.getApplicationName(module))
+                                                                                                .build());
+        Map<String, Object> parameters = (Map<String, Object>) resource.getParameters()
+                                                                       .getOrDefault(SupportedParameters.SERVICE_CONFIG,
+                                                                                     Collections.emptyMap());
+        if (parameters == null) {
```

Comment:
Parameters can't be null since it's `.getOrDefault(..., Collections.emptyMap())`